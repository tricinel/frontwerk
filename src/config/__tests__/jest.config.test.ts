/* eslint-disable global-require */
import jestSerializerPath from 'jest-serializer-path';

import {
  unquoteSerializer,
  winPathSerializer
} from '../../helpers/serializers';

expect.addSnapshotSerializer(unquoteSerializer);
expect.addSnapshotSerializer(winPathSerializer);
expect.addSnapshotSerializer(jestSerializerPath);

jest.mock('../../utils/fromRoot', () => () => '');

afterEach(() => {
  jest.resetModules();
});

describe('Gets the correct test environment for jest', () => {
  test('with jsdom', () => {
    jest.mock('../../utils/hasDep', () => ({
      hasDep: (dep: string): boolean => dep === 'rollup' || dep === 'react'
    }));
    const config = require('../jest.config');
    expect(config.testEnvironment).toEqual('jsdom');
  });

  test('with node', () => {
    jest.mock('../../utils/hasDep', () => ({
      hasDep: () => false
    }));
    const config = require('../jest.config');
    expect(config.testEnvironment).toEqual('node');
  });
});

describe('Sets the transform property on the jest config', () => {
  test('Uses the babel transform', () => {
    jest.mock('../../utils/fileExists', () => () => false);
    const config = require('../jest.config');
    expect(config.transform).toMatchSnapshot();
  });

  test('Does not use the babel transform', () => {
    jest.mock('../../utils/fileExists', () => () => true);
    const config = require('../jest.config');
    expect(config.transform).toMatchSnapshot();
  });
});
