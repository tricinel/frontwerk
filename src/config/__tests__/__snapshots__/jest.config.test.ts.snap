// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets the transform property on the jest config Does not use the babel transform 1`] = `undefined`;

exports[`Sets the transform property on the jest config Uses the babel transform 1`] = `
Object {
  ^.+/.[t|j]sx?$: babel-jest,
}
`;
